crates/pathfinder/src/monitoring/metrics.rs

```diff
@@ -0,0 +1,80 @@
+pub mod middleware {
+    use jsonrpsee::core::middleware::Middleware;
+    use metrics::increment_counter;
+    use std::time::Instant;
+
+    #[derive(Debug, Clone)]
+    pub struct RpcMetricsMiddleware;
+
+    impl Middleware for RpcMetricsMiddleware {
+        type Instant = Instant;
+
+        fn on_request(&self) -> Self::Instant {
+            Instant::now()
+        }
+
+        fn on_call(&self, name: &str) {
+            increment_counter!(format!("{name}_calls_total"));
+        }
+    }
+
+    #[derive(Debug, Clone)]
+    pub enum MaybeRpcMetricsMiddleware {
+        Middleware(RpcMetricsMiddleware),
+        Noop,
```

Comment:
Minor nit. I agree the suggestion is more annoying to type.
```suggestion
        NoOp,
```

Reply:
I think the noop is the more traditional name. Though I guess overall here an `Option<RpcMetricsMiddleware>` could had worked just as well?

Reply:
I think the `Option` would need to be new typed since it needs an `impl` later on.

Reply:
I'm not sure if I tried impl Recorder for Option<RpcMetricsMiddleware> yet, lemme try it .

Reply:
Unfortunately I get [this error](https://doc.rust-lang.org/error-index.html#E0117):
```
error[E0117]: only traits defined in the current crate can be implemented for types defined outside of the crate
  --> crates/pathfinder/src/monitoring/metrics.rs:57:5
   |
57 |       impl<M> jsonrpsee::core::middleware::Middleware for Option<M>
   |       ^                                                   --------- `std::option::Option` is not defined in the current crate
   |  _____|
   | |
58 | |     where
59 | |         M: jsonrpsee::core::middleware::Middleware,
60 | |     {
...  |
72 | |         }
73 | |     }
   | |_____^ impl doesn't use only types from inside the current crate
   |
   = note: define and implement a trait or new type instead
```
This works ofc:
```
    trait CrateLocalTrait {}

    impl<M> CrateLocalTrait for Option<M> where M: jsonrpsee::core::middleware::Middleware {}
```
Adding a `NewType(Option<RpcMetricsMiddleware>)` is similar to a new enum, so I'll just rename the no-op variant.